sklearn/neighbors/base.py

```diff
@@ -22,8 +22,10 @@
 from ..metrics import pairwise_distances_chunked
 from ..metrics.pairwise import PAIRWISE_DISTANCE_FUNCTIONS
 from ..utils import check_X_y, check_array, gen_even_slices
+from ..metrics.pairwise import _NAN_METRICS
```

Comment:
We no longer need the changes to Nearest Neighbors for KNNImputer to work. Do you think they are otherwise valuable?

Reply:
I think there is minimum value in having Nearest Neighbors support nan metrics.

Reply:
Not quite sure what you're proposing here.

Reply:
One motivation by including the "masked euclidean distance" metric was to add it as a metric to Nearest Neighbors. Now that these changes are not needed in Nearest Neighbors, I think it would be best to add the KNNImputer back into this PR.

Reply:
Feel free to try merge the two PRs!